examples/fedrag/pyproject.toml

```diff
@@ -0,0 +1,62 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "fedrag"
+version = "1.0.0"
+description = ""
+license = "Apache-2.0"
+dependencies = [
+    "flwr[simulation]>=1.15.2",
+    "numpy>=1.21.0",
+    "faiss-cpu>=1.10.0",
+    "tqdm>=4.67",
+    "sentence_transformers>=3.4.1",
+]
+
+[tool.hatch.build.targets.wheel]
+packages = ["."]
+
+[tool.flwr.app]
+publisher = "dimitris"
+
+[tool.flwr.app.components]
+serverapp = "fedrag.server_app:app"
+clientapp = "fedrag.client_app:app"
+
+[tool.flwr.app.config]
+# Possible question answer datasets: PubMedQA, BioASQ, MMLU, MedQA, MedMCQA
+# you can pass them as follows: "pubmedqa|bioasq|mmlu|medqa|medmcqa"
+# the vertical bar separates the different benchmark datasets
+server-qa-datasets = "pubmedqa|bioasq"
+# force to only evaluate the first 100 questions from each benchamrk
+# dataset, if this value is not set, then evaluate all questions
+server-qa-num = 10
+# Pass corpus as "StatPearls|Textbooks", the corpus will be assigned
+# to each client participating in the federation in a Round-Robin fashion
+# The supported corpora are:
+# PubMed (23.9M docs), StatPearls (9.3k docs), Textbooks (18 docs), Wikipedia (6.5M docs)
+# you can pass them as follows: "pubmed|statpearls|textbooks|wikipedia"
+# the vertical bar separates the different corpus
+clients-corpus-names = "statpearls|textbooks"
+# The value of the Reciprocal Rank Fusion. The value will be used
+# to merge multiple documents together at the server-side
+# If the value is not set then only the retrieval scores are considered
+k-rrf = 60
+# The k-nearest-neighbor (most relevant) documents to the given search query
+# used by clients to retrieve top-k relevant documents and at the server
+# to select the final top-k documents after the merge operation
+k-nn = 8
+# The LLM's HuggingFace path that will be used to for the RAG query
+# e.g., "meta-llama/Llama-3.2-1B-Instruct" or "meta-llama/Llama-3.2-3B-Instruct"
+# We use HF's SmolLM model because for Llama we need first to accept the terms
+server-llm-hfpath = "HuggingFaceTB/SmolLM2-1.7B-Instruct"
+
+
+[tool.flwr.federations]
+default = "local-simulation"
+
+[tool.flwr.federations.local-simulation]
+options.num-supernodes = 2
+options.backend.client-resources.num-cpus = 2
```

Comment:
Given that this is the default, should we keep it? I'd keep it if we also add the line for `num-gpus` and set it to `0`. And then include in the `README.md` what to change so the example runs with gpu

Reply:
does the example run with GPU also?